src/d3a/models/area/event_dispatcher.py

```diff
@@ -159,12 +171,23 @@ def broadcast_notification(
         # Broadcast to children in random order to ensure fairness
         for child in sorted(self.area.children, key=lambda _: random()):
             child.dispatcher.event_listener(event_type, **kwargs)
-        self._broadcast_notification_to_area_and_child_agents(
-            AvailableMarketTypes.SPOT, event_type, **kwargs)
-        self._broadcast_notification_to_area_and_child_agents(
-            AvailableMarketTypes.BALANCING, event_type, **kwargs)
-        self._broadcast_notification_to_area_and_child_agents(
-            AvailableMarketTypes.SETTLEMENT, event_type, **kwargs)
+
+        market_id = kwargs.get("market_id")
+        if not market_id and isinstance(event_type, MarketEvent):
+            assert False, "MarketEvent should always provide a market_id."
+
+        if isinstance(event_type, AreaEvent) or self.area.is_market_spot(market_id):
+            self._broadcast_notification_to_area_and_child_agents(
+                AvailableMarketTypes.SPOT, event_type, **kwargs)
+        if isinstance(event_type, AreaEvent) or self.area.is_market_balancing(market_id):
+            self._broadcast_notification_to_area_and_child_agents(
+                AvailableMarketTypes.BALANCING, event_type, **kwargs)
+        if isinstance(event_type, AreaEvent) or self.area.is_market_settlement(market_id):
+            self._broadcast_notification_to_area_and_child_agents(
+                AvailableMarketTypes.SETTLEMENT, event_type, **kwargs)
+        if isinstance(event_type, AreaEvent) or self.area.is_market_future(market_id):
+            self._broadcast_notification_to_area_and_child_agents(
+                AvailableMarketTypes.FUTURE, event_type, **kwargs)
```

Comment:
`_broadcast_notification_to_area_and_child_agents` receives two types of events: `MarketEvent, AreaEvent`. For `AreaEvent`s (TICK, MARKET_CYCLE, ..), we want to dispatch it to all of the connected areas independent of the type. AFAIU, this is only temporary, because ideally, we have AreaEvents for each market type (like the `BALANCING_MARKET_CYCLE`) . Because the area may want to react differently if a FutureMarket TICK event is received rather then a normal TICK is received. Shall we add a story for that @spyrostz ?

In order to answer your question, the second condition is for the `MarketEvents` like `OFFER`, `BID`, `BID_TRADED` etc. These should only be dispachted to the corresponding IAA for the market type. For example, a Settlement IAA is not interested, if an OFFER is posted in the spot market. 
Does it make sense? BTW, I took a lot of time to come up with a more DRY version of this, but could not find one.

Reply:
@hannesdiedrich feel free to add a story for that, however if we decide to move forward to organic grid structure (where we will not have area dispatcher but a central entity that keeps track of the grid structure and dispatches events to different markets or strategies - if we implement that soon enough then this ticket will not really be useful). Having said all the above, since I am not really certain how far in the future will we implement that, feel free to open a story to improve the situation in the short term